apps/prairielearn/python/prairielearn/core.py

```diff
@@ -845,17 +845,7 @@ def get_color_attrib(
 
 
 # This internal represents most the types that would pass a np.isscalar check
-_NumPyScalarType = (
-    np.complex64
-    | np.complex128
-    | bool
-    | int
-    | float
-    | complex
-    | str
-    | bytes
-    | np.generic
-)
+_NumPyScalarType = bool | int | float | complex | str | bytes | np.generic
```

Comment:
AFAICT `np.complex64` and `np.complex128` are subtypes of `np.generic`, so explicitly listing them separately wasn't needed.

Reply:
I do wonder if we want to include `str` and `bytes` here. What do we gain from that? Seems like it would just mislead callers since we reject those types at runtime? And is there any reason to include `bool` here, or indeed to list `int`/`float`/`complex` separately instead of using `numbers.Number`?

Reply:
This type was intended to line up with the type guard associated with `np.isscalar`. I am fine changing the type as long as it is renamed. I also agree that the current type isn't ideal; and can mislead callers.

Reply:
I changed the type to `_NumericScalarType` in 1d83004. It's now defined as follows:

```py
_NumericScalarType = numbers.Number | complex | np.generic
```

Even though `isinstance(complex(1, 2), numbers.Number) == True`, pyright doesn't seem to recognize `complex` as assignable to `numbers.Number`, so listing `complex` separately was necessary.